lib/mayaUsd/resources/ae/usdschemabase/attribute_custom_control.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2023 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import mayaUsd.lib
+
+from maya.OpenMaya import MGlobal
+
+import os.path
+import json
+
+
+def getNiceAttributeName(ufeAttr, attrName):
+    '''
+    Convert the attribute name into  nice name.
+    '''
+    # Note: the uiname metadata comes from LookdevX and was used for connections.
+    if hasattr(ufeAttr, 'displayName'):
+        attrName = ufeAttr.displayName
+    elif hasattr(ufeAttr, "hasMetadata") and ufeAttr.hasMetadata("uiname"):
```

Comment:
Can this not be addressed on LookdevX end, by setting the displayName instead?

It seems to me that this is client's business logic.

Reply:
I believe the uiname is part of MaterialX specs. Either way, I did not want to break backward compatibility.